coil-network/src/jvmCommonMain/kotlin/coil3/network/utils.jvm.kt

```diff
@@ -14,8 +14,9 @@ internal actual suspend fun ByteReadChannel.writeTo(sink: BufferedSink) {
 internal actual suspend fun ByteReadChannel.writeTo(fileSystem: FileSystem, path: Path) {
     if (fileSystem === FileSystem.SYSTEM) {
         // Fast path: normal jvm File, write to FileChannel directly.
-        RandomAccessFile(path.toFile(), "rw").use {
-            copyTo(it.channel)
+        RandomAccessFile(path.toFile(), "rw").use { file ->
+            val copied = copyTo(file.channel)
+            file.setLength(copied) // truncate tail that could remain from the previously written data
```

Comment:
What happens if we throw during `copyTo`? Should we set the length to 0?

Reply:
DiskCache should abort edit and delete that entry
https://github.com/coil-kt/coil/blob/8389ba8b95a41b49d2d347aaf6ea5abb7f3eff90/coil-network/src/commonMain/kotlin/coil3/fetch/NetworkFetcher.kt#L166